src/inverse-kinematics/CMakeLists.txt

```diff
@@ -11,18 +11,22 @@ set(IDYN_TREE_IK_HEADERS include/iDynTree/ConvexHullHelpers.h
                          include/iDynTree/BoundingBoxHelpers.h
                          include/iDynTree/InverseKinematics.h)
 
-set(PRIVATE_IDYN_TREE_IK_SOURCES src/InverseKinematicsNLP.cpp
-                                 src/InverseKinematicsData.cpp
-                                 src/TransformConstraint.cpp)
-set(PRIVATE_IDYN_TREE_IK_HEADERS include/private/InverseKinematicsNLP.h
-                                 include/private/InverseKinematicsData.h
-                                 include/private/TransformConstraint.h)
-
-source_group("Private\\Header Files" FILES ${PRIVATE_IDYN_TREE_IK_HEADERS})
-source_group("Private\\Source Files" FILES ${PRIVATE_IDYN_TREE_IK_SOURCES})
-
-add_library(${libraryname} ${IDYN_TREE_IK_HEADERS} ${IDYN_TREE_IK_SOURCES}
-                           ${PRIVATE_IDYN_TREE_IK_SOURCES} ${PRIVATE_IDYN_TREE_IK_HEADERS})
+if(IDYNTREE_USES_IPOPT)
```

Comment:
I had this `if` since those classes are not compiled in the dummy class.
@traversaro please check if it looks sound for you or if you think there is a better way to handle it.

Reply:
Using `target_sources` you can avoid to duplicate the `add_library` call, in any case what you did is ok, thanks!